### PR TITLE
Update versions of VS dependecies

### PR DESCRIPTION
**Pull request by genlu**
The new version of those packages are now targeting netstandard2.0. I'm queuing [RPS validation](https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/268908) for this change. 

With this, we are down to ~~3~~ 2 references of remote project not supporting netstandard or netcoreapp

~~Microsoft.VisualStudio.Imaging~~ (turns out this reference is unused)

Microsoft.CodeAnalysis.Elfie:
It seems we own this. The code is at https://github.com/microsoft/elfie-arriba
Talked to someone who worked on Elfie, it sounds like there's no automated infrastructure for build and publish. Worst case we will need to convert it to SDK style project and retarget it ourselves, and manually submit it for codesign.

Microsoft.VisualStudio.Language.Intellisense
Talked to @AmadeusW, this package can't target netstandard2.0 or netcoreapp3.1 due to WPF dependencies (also he suggested most of the types we relies on from this package do indeed requires WPF, so refactoring the package might not help either). I'm trying to see if I can move things depends on this package from EditorFeatures up to EditorFeatures.Wpf. On the other hand, given that we are likely make the servicehub core switch in 16.9, when .Net 5 already GA'd, it might be enough for them to just target net5 w/o any change in Roslyn. This means our remote will be targeting .net5, thus services running in our remote process (intellicode, unitesting and razor) will need to multi-targeting net472 and net5 too.

FYI @tmat @sharwell

### CODE REVIEW

File: src/Workspaces/Remote/Core/RemoteEndPoint.cs [hunk 1]
```diff
@@ -390,7 +390,7 @@ private void LogDisconnectInfo(JsonRpcDisconnectedEventArgs? e)
         {
             if (e != null)
             {
-                LogError($@"Stream disconnected unexpectedly:  {e.Reason}, '{e.Description}', LastMessage: {e.LastMessage}, Exception: {e.Exception?.Message}");
```
Comment (Cosifne) [inline src/Workspaces/Remote/Core/RemoteEndPoint.cs:393]: Why the LastMessage is removed here?

File: src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj [hunk 2]
```diff
@@ -35,10 +35,11 @@
     <PackageReference Include="Microsoft.VisualStudio.Threading" Version="$(MicrosoftVisualStudioThreadingVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.SDK.EmbedInteropTypes" Version="$(MicrosoftVisualStudioSDKEmbedInteropTypesVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.ImageCatalog" Version="$(MicrosoftVisualStudioImageCatalogVersion)" />
-    <PackageReference Include="Microsoft.VisualStudio.Imaging" Version="$(MicrosoftVisualStudioImagingVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Imaging.Interop.14.0.DesignTime" Version="$(MicrosoftVisualStudioImagingInterop140DesignTimeVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Language" Version="$(MicrosoftVisualStudioLanguageVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Language.Intellisense" Version="$(MicrosoftVisualStudioLanguageIntellisenseVersion)" />
+    <PackageReference Include="Microsoft.ServiceHub.Client" Version="$(MicrosoftServiceHubClientVersion)" />
```
Comment (Cosifne) [inline src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:41]: Why editor feature needs to reference Servicehub package? Is it used for version conflicts?

Comment (genlu) [reply src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:41]: Yes, it's referenced by one of the packages from VS.

File: src/Workspaces/Remote/Core/RemoteEndPoint.cs (hunk 1 above)
Comment (genlu) [reply src/Workspaces/Remote/Core/RemoteEndPoint.cs:393]: it's obsolete in new version.

File: eng/Versions.props [hunk 3]
```diff
@@ -253,7 +253,7 @@
       create a test insertion in Visual Studio to validate.
     -->
     <NewtonsoftJsonVersion>12.0.2</NewtonsoftJsonVersion>
-    <StreamJsonRpcVersion>2.4.34</StreamJsonRpcVersion>
+    <StreamJsonRpcVersion>2.6.21-alpha</StreamJsonRpcVersion>
```
Comment (dibarbet) [inline eng/Versions.props:256]: +1 to Shen's comment - we should do a test insertion for this.

Comment (genlu) [reply eng/Versions.props:256]: RPS has no regressions (other than existing one of loading wer.dll, caused by https://github.com/dotnet/roslyn/pull/46703)

**Pull request merged**